docs: Clarification and edits in Data availability (#949)
starknet-io/starknet-docs: components/Starknet/modules/architecture_and_concepts/pages/Network_Architecture/on-chain-data.adoc

[id="introduction"]
== Introduction

> In the current stage of the Alpha, Starknet operates in Validity Rollup mode, also referred to as ZK Rollup mode. This means that upon the acceptance of a state update onchain, the state diff between the previous and new state is sent as calldata to Ethereum.

Author: ```suggestion
In the current stage of the Alpha, Starknet operates in Validity Rollup mode, also referred to as ZK-Rollup mode. This means that upon the acceptance of a state update onchain, the state diff between the previous and new state is sent as calldata to Ethereum.
```

This data allows anyone that observes Ethereum to reconstruct the current state of Starknet.

[NOTE]
====
To update the Starknet state on L1, it suffices to send a valid proof, without information
on the transactions or particular changes that this update caused.

Consequently, more information must be provided in order to allow other parties to locally track Starknet's state.
====

== Data availability: v0.11.0 and forward

[id="v0.11.0_format"]
=== v0.11.0 format

The state diffs contain information on every contract whose storage was updated and additional information on contract deployments.

> For each affected contract the following information is sent to the Ethereum core contract in a transaction as calldata:

Author: ```suggestion
For each affected contract, the following information is sent as calldata on L1:
```

]
----

* The first element, `1`, is the number of contracts whose state was updated.
* The second element, `2019172390095051323869047481075102003731246132997057518965927979101413600827`, is the address of the first, and only, contract whose state changed.
* The third element, `18446744073709551617`, which is 2^64^+1, encodes the following:
** The class information flag is `0`, that is, the contract was not deployed or replaced just now, so you shouldn't treat the next word as the class hash.
** The new nonce is `1`.
** One storage cell was updated.
* The next two elements, `100` and `200`, encode the storage update, where the value of key `100` is set to `200`.

> * he new declare section: `1` includes a single xref:Network_Architecture/transactions.adoc#declare_v2[declare v2] transaction in this state update, and the next two elements

Author: ```suggestion
* the new declare section: `1` includes a single xref:Network_Architecture/transactions.adoc#declare_v2[declare v2] transaction in this state update, and the next two elements
```
